src/services/http_request_service.ts

```diff
@@ -28,6 +28,44 @@ export default class HttpRequestService {
   // FILE MARKER - PUBLIC //////////////////////////////////////////////////////
   //////////////////////////////////////////////////////////////////////////////
 
+  /**
+   * @description
+   *     Checks if the incoming request accepts the type(s) in the parameter.
+   *     This method will check if the requests `Accept` header contains
+   *     the passed in types
+   *
+   * @param string|string[] type
+   *     The content-type/mime-type(s) to check if the request accepts it
+   *
+   * @example
+   *     // YourResource.ts - assume the request accepts "text/html"
+   *     const isAccepted = this.request.accepts("text/html"); // "text/html"
+   *     // or can also pass in an array and will match on the first one found
+   *     const isAccepted = this.request.accepts(["text/html", "text/xml"]); // "text/html"
+   *     // and will return false if not found
+   *     const isAccepted = this.request.accepts("text/xml"); // false
+   *
+   * @return boolean|string
+   *     False if the request doesn't accept any of the passed in types,
+   *     or the content type that was matches
+   */
+  public accepts (request: any, type: string|string[]): boolean|string {
+    const acceptHeader = request.headers.get('Accept')
+    const acceptedContentTypes = acceptHeader.split(';')[0];
+
+    // for when `type` is a string
+    if (typeof type === 'string')
+      return acceptedContentTypes.indexOf(type) >= 0 ? type : false;
+
+    // for when `type` is an array
+    const matches = type.filter(t => acceptedContentTypes.indexOf(t) >= 0);
```

Comment:
I would swap `.filter()` for `.find()`.
```typescript
const matches = type.find(t => acceptedContentTypes.indexOf(t) >= 0); // This contains the first element found.
return typeof matches != "undefined" ? matches : false;
```

Reply:
I don't think i see much difference in using `.find()`, but I have incorporated part of your approach to make that block of code cleaner (yet to be committed) but looks like:
```
const matches = type.fnd(...); // the usual
return matches.length ? macthes[0] : false;
```
Cleaner code than what was before:
```
if (something)
  if (something)
     return something
return something else
```

Reply:
Your implementation is fine, honestly. Mine doesn't address any issue, just code style, since you are actually trying to find ONE element.